### PR TITLE
Remove master profile and replace with property override

### PR DESCRIPTION
**Pull request by pivovarit**

### CODE REVIEW

**Review by enozcan — COMMENTED** [review]
Why don't we just change the version of master profile?

**Comment by hasancelik** [comment]
> Why don't we just change the version of master profile?

@enozcan @pivovarit actually, we do not need `master` profile at pom anymore. It was just used to test codebase against IMDG master. I have already [removed it from kubernetes repo](https://github.com/hazelcast/hazelcast-kubernetes/pull/283/files).

**Comment by pivovarit** [comment]
To be honest, I did not even realize the existence of the _master_ profile, but I question the need for it since you can achieve the same by just swapping the dependency version without hardcoding anything into the code itself.

So, I just fixed a broken pipeline definition but the next stage would be to remove the _master_ profile

**Review by hasancelik — APPROVED** [review]

File: .github/workflows/build-snapshot.yml [hunk 1]
```diff
@@ -26,5 +27,5 @@ jobs:
           restore-keys: ${{ runner.os }}-maven-
 
       - name: Build with Maven
-        run: mvn verify -Pmaster
+        run: mvn -Dhazelcast.version=${{ matrix.hazelcast }} verify
```
Comment (enozcan) [inline .github/workflows/build-snapshot.yml:30]: idk if test runners disable it but if not, we better add `-Dhazelcast.phone.home.enabled=false`

**Review by enozcan — APPROVED** [review]
LGTM with a minor comment. Also jenkins build will fail if there is any and hence needs to be disabled.

File: .github/workflows/build-snapshot.yml (hunk 1 above)
Comment (hasancelik) [reply .github/workflows/build-snapshot.yml:30]: it is already disabled by default at surefire plugin --> https://github.com/hazelcast/hazelcast-zookeeper/blob/master/pom.xml#L191

**Review by hasancelik — APPROVED** [review]

**Comment by pivovarit** [comment]
@enozcan I killed them long time ago

**Pull request merged**